pages/forms/TripForm/[id].js

```diff
@@ -0,0 +1,83 @@
+import CancelButton from "@/components/CancelButton";
+import CreateButton from "@/components/CreateButton";
+import { useRouter } from "next/router";
+import useSWR from "swr";
+
+const fetcher = (...args) => fetch(...args).then((res) => res.json());
+
+export default function TripDetailsForm() {
+  const { mutate } = useSWR("/api/trips");
+  const router = useRouter();
+  const { id } = router.query;
+  const { data: trips, isLoading } = useSWR(
+    id ? `/api/trips/${id}` : null,
+    fetcher
+  );
+
+  if (!trips || isLoading) {
+    return <>is Loading...</>;
+  }
+  const { title, location, timePeriod, img } = trips;
+  const country = location.map((location) => `${location.country}`);
+  const city = location.map((location) => `${location.city}`);
+  const startDate = timePeriod.map((timePeriod) => `${timePeriod.startDate}`);
+  const endDate = timePeriod.map((timePeriod) => `${timePeriod.endDate}`);
+
+  async function handleEditTrip(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const tripData = Object.fromEntries(formData);
+
+    const editedTrip = {
+      title: tripData.title,
+      location: [{ country: tripData.country, city: tripData.city }],
+      timePeriod: [
+        { startDate: tripData.startDate, endDate: tripData.endDate },
+      ],
+    };
+
+    const response = await fetch(`/api/trips/${id}`, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(editedTrip),
+    });
+
+    if (!response.ok) {
+      console.error(response.status);
+      return;
+    }
+
+    mutate();
+    router.push(`/trip/${id}`);
+  }
+
+  return (
+    <form onSubmit={handleEditTrip}>
+      <label>
+        Country
+        <input name="country" defaultValue={country} minLength={3} required />
+      </label>
+      <label>
+        City
+        <input name="city" defaultValue={city} />
+      </label>
+      <label>
+        Title
+        <input name="title" defaultValue={title} minLength={3} required />
+      </label>
+      <label>
+        Start date
+        <input name="startDate" type="date" defaultValue={startDate} required />
+      </label>
+      <label>
+        End date
+        <input name="endDate" type="date" defaultValue={endDate} />
+      </label>
+      <CreateButton>Save</CreateButton>
```

Comment:
It's a good idea to make the button role clear, e.g., using type="submit" for the Save button.
```suggestion
      <CreateButton type="submit">Save</CreateButton>
```

🤓 **Additionally,** I would not nest the `ClancelButton`  (which is actually a link that leads back to the homepage, right?)

Reply:
The type of the button is already set within the CreateButton component, so we thought it would be redundant to put it there, right? 

```
import { StyledCreateButton } from "./CreateButton.styled";

export default function CreateButton({ children }) {
  return <StyledCreateButton type="submit">{children}</StyledCreateButton>;
}
```

Also, could you please elaborate further on the nesting of the CancelButton? We don't quite understand your feedback 🤔